### DISCUSSION TITLE
launch.json: Cannot launch program from $PATH

### DISCUSSION THREAD

**Discussion by jvallesf** [microsoft/vscode-cpptools#8400] [General]
Hello,

I am working with a Node.js C++ addon and struggling to set up launch.json to debug C++ and Javascript at the same time. On Windows I have already done it with the debugger "cppvsdbg" launching the program "node" (node.exe, in $PATH).

As you can see in the launch.json configuration below, on Linux I am forced to add a special configuration, first of all for the different debugger "cppdbg", which is ok, but also for the program to launch, in this case "${workspaceRoot}/node", which is a link I created to the node executable, because it does not recognize "program": "node", even though it is also in $PATH.

If I try to launch "program": "node" in Linux I get a dialog with the following error: "launch: program "node" does not exist".

Do you know why "program": "node" is not recognized as an executable in Linux? I have already checked that `$ node` works in the terminal.

It would be great if there was some workaround, because we could have a uniform configuration for both Linux and Windows, and avoid creating the link to the node executable.

Thank you!

```
{
    "name": "test-cpp-debug",
    "type": "cppdbg",
    "request": "launch",
    "program": "node",
    "args": [
        "--expose-gc",
        "--inspect=9229",
        "${workspaceFolder}/test/tests.js",
        "--dev",
        "--deb"
    ],
    "stopAtEntry": false,
    "cwd": "${workspaceRoot}",
    "console": "integratedTerminal",
    "windows": {
        "type": "cppvsdbg",
        "program": "node",
    },
    "linux": {
        "type": "cppdbg",
        "program": "${workspaceRoot}/node"
    }
}
// This configuration works for Win and Linux, but notice the "program": "${workspaceRoot}/node" in the Linux configuration,
// which is an ad hoc link to the Node executable in Linux.
```

**Comment by bobbrow** [comment]
@WardenGnaw is this something that needs to be fixed in our debug provider implementation, or is it a VS Code issue? (i.e. do you just pass this info down to them?)

**Reply by WardenGnaw** [reply]
We expect the full program path in `MIEngine`, `vsdbg` may be doing something different. 

See https://cs.github.com/microsoft/MIEngine/blob/d8e84841c31b11db3cf767a3794f0972a349299a/src/OpenDebugAD7/AD7DebugSession.cs#L172
